README.md

```diff
@@ -23,49 +23,61 @@ for more information regarding Astarte and the available SDKs.
 
 ## Basic usage
 
-### Create a device
-Initializing an instance of a device can be performed in three steps, as seen below.
-```Rust
+```rust
 use astarte_device_sdk::{
     database::AstarteSqliteDatabase,
     options::AstarteOptions,
-    AstarteDeviceSdk};
-
-// (optional) Initialize a database to store the properties
-let db = AstarteSqliteDatabase::new("sqlite::memory:").await.unwrap();
-
-// Initialize device options
-let sdk_options = AstarteOptions::new(&realm, &device_id, &credentials_secret, &pairing_url)
-    .interface_directory("./examples/interfaces")
-    .unwrap()
-    .database(db);
-
-// Create the device instance
-let mut device = AstarteDeviceSdk::new(&sdk_options).await.unwrap();
-```
-### Publish data from device
-Publishing new values can be performed using the `send` and `send_object` functions.
-```Rust
-// Send individual datastream or set individual property
-let data: i32 = 12;
-device.send("interface.name", "/endpoint/path", data).await.unwrap();
-
-// Send aggregated object datastream
-#[derive(Debug, AstarteAggregate)]
-struct MyAggObj {
-    endpoint1: f64,
-    endpoint2: i32
-}
-let data = MyAggObj {endpoint1: 1.34, endpoint2: 22};
-device.send_object("interface.name", "/common/endpoint/path", data).await.unwrap();
-```
-### Receive a server publish
-Polling for new data can be performed using the function `handle_events`.
-```Rust
- loop {
-    match device.handle_events().await {
-        Ok(data) => (), // Handle data
-        Err(err) => (), // Handle errors
+    AstarteDeviceSdk,
+    AstarteError
+};
+
+async fn run_astarte_device() -> Result<(), AstarteError> {
+    
+    let realm = "realm_name";
+    let device_id = "device_id";
+    let credentials_secret = "device_credentials_secret";
+    let pairing_url = "astarte_cluster_pairing_url";
+
+    // Initializing an instance of a device can be performed in three steps, as seen below.
+    
+    // 1. (optional) Initialize a database to store the properties
+    let db = AstarteSqliteDatabase::new("sqlite::memory:").await?;
+    
+    // 2. Initialize device options
```

Comment:
// 2. Initialize device options (the ".database(db)" is not needed if 1 was skipped)